core/src/main/java/jenkins/model/identity/IdentityRootAction.java

```diff
@@ -0,0 +1,81 @@
+package jenkins.model.identity;
+
+import hudson.Extension;
+import hudson.model.UnprotectedRootAction;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import org.apache.commons.codec.Charsets;
+import org.apache.commons.codec.binary.Base64;
+import org.jenkinsci.remoting.util.KeyUtils;
+
+/**
+ * A simple root action that exposes the public key to users so that they do not need to search for the
+ * {@code X-Instance-Identity} response header, also exposes the fingerprint of the public key so that people
+ * can verify a fingerprint of a master before connecting to it.
+ *
+ * @since FIXME
+ */
+@Extension
+public class IdentityRootAction implements UnprotectedRootAction {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDisplayName() {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getUrlName() {
+        return InstanceIdentityProvider.get(RSAPrivateKey.class) == null ? null : "instance-identity";
+    }
+
+    /**
+     * Returns the PEM encoded public key.
+     *
+     * @return the PEM encoded public key.
+     */
+    public String getPublicKey() {
```

Comment:
Yeah I need to get the corresponding changes in instance-identity merged and released first to get the test in place